airbyte_cdk/cli/airbyte_cdk/_secrets.py

```diff
@@ -1,36 +1,71 @@
 # Copyright (c) 2025 Airbyte, Inc., all rights reserved.
-"""Secret management commands.
+"""**Secret management commands.**
 
 This module provides commands for managing secrets for Airbyte connectors.
 
-Usage:
-    airbyte-cdk secrets fetch --connector-name source-github
-    airbyte-cdk secrets fetch --connector-directory /path/to/connector
-    airbyte-cdk secrets fetch  # Run from within a connector directory
+**Usage:**
 
-Usage without pre-installing (stateless):
-    pipx run airbyte-cdk secrets fetch ...
-    uvx airbyte-cdk secrets fetch ...
+```bash
+# Fetch secrets
+airbyte-cdk secrets fetch --connector-name source-github
+airbyte-cdk secrets fetch --connector-directory /path/to/connector
+airbyte-cdk secrets fetch  # Run from within a connector directory
 
-The 'fetch' command retrieves secrets from Google Secret Manager based on connector
+# List secrets (without fetching)
+airbyte-cdk secrets list --connector-name source-github
+airbyte-cdk secrets list --connector-directory /path/to/connector
+```
+
+**Usage without pre-installing (stateless):**
+
+```bash
+pipx run airbyte-cdk secrets fetch ...
+uvx airbyte-cdk secrets fetch ...
+```
+
+The command retrieves secrets from Google Secret Manager based on connector
 labels and writes them to the connector's `secrets` directory.
 """
 
+from __future__ import annotations
+
 import json
+import logging
 import os
+from functools import lru_cache
 from pathlib import Path
+from typing import Any, cast
 
+import requests
 import rich_click as click
+import yaml
+from click import style
+from rich.console import Console
+from rich.table import Table
 
-from airbyte_cdk.cli.airbyte_cdk._util import resolve_connector_name_and_directory
+from airbyte_cdk.cli.airbyte_cdk._util import (
+    resolve_connector_name,
+    resolve_connector_name_and_directory,
+)
 
 AIRBYTE_INTERNAL_GCP_PROJECT = "dataline-integration-testing"
```

Comment:
Should this be an input argument, rather than hardcoded? + nit: we expose the names we don't want, i think.

Reply:
As shared in DM, we already do have this in tons of places in the airbyte repo. Keeping as-is for now, as it allows devs to immediately get rolling. In future, I'd like to parametrize this or scrape from env vars, as we do with GCP_GSM_CREDENTIALS.